src/packaging/markers.py

```diff
@@ -231,7 +316,7 @@ def __eq__(self, other: Any) -> bool:
 
         return str(self) == str(other)
 
-    def evaluate(self, environment: Optional[Dict[str, str]] = None) -> bool:
+    def evaluate(self, environment: Optional[Environment] = None) -> bool:
```

Comment:
This isn't actually correct thanks to line 331. Could you please revert this and the doc change for it?

Reply:
Gotcha. Is the `environment` parameter here expected to be an arbitrary dictionary then?

If that's the case then the annotation in `_evaluate_markers` was also incorrect.

5dca39a3cf2a0ab019cac7ae44a021f8665c95c2